### ISSUE TITLE
[Support] Parallelize loops with constant values as loop-bounds 

### ISSUE THREAD

**Issue by jjfumero** [beehive-lab/TornadoVM#21]
Currently, TornadoVM only parallelizes loops in which loop bounds are taken as array-length.
Add support for constant-values. 

```java
for (@Parallel i = 0; i < CONSTANT; i++) {
   ... 
}
```
To support this, the compiler phases `TornadoAutoParalleliser` and `TornadoParallelScheduler` need to be updated. 
This change also influences the batch processing, in which loop-bounds need to be updated according to the batch size.
This change will also provide a set of unittests to check this feature.

**Comment by jjfumero** [comment]
The issue was that for loops such as:

```java
for (@Parallel int i = 0; i <= n; i++) {
      a[i] = 10;
}
```

The constant value is evaluated and rescheduled. The `TornadoTaskSpecialization` expects the loop bounds attached to the `ParallelRangeNode` instead of the conditional node. 

The proposed change is as follows:

![Screenshot from 2020-06-25 12-29-46](https://user-images.githubusercontent.com/5602565/85703480-b04cb200-b6df-11ea-93c1-9d165b86a205.png)


PR coming soon.

**Comment by jjfumero** [comment]
Fixed in #22
